### PR TITLE
Hide Mobile Menu When Menu is Empty and Menu Search is Disabled

### PR DESCRIPTION
**Pull request by AlexGStapleton**
This PR will hide the mobile menu when the menu is empty _and_ the Menu Search is disabled. Previously the mobile menu would show but it would be completely empty.

### CODE REVIEW

**Comment by Misplon** [comment]
Thanks :)

* If only the mini-cart is displayed the mobile menu doesn't display. Ideally, it needs to display like it does when only search is shown.

* If the search and mini-cart are displayed, nothing else, the cart shows twice. This is present in develop but can be fixed now in this PR.

File: header.php [hunk 1]
```diff
@@ -85,15 +85,21 @@
...
 
-								<a href="#menu" id="mobile-menu-button">
-									<?php siteorigin_north_display_icon( 'menu' ) ?>
-									<?php if ( siteorigin_setting( 'responsive_menu_text' ) ) : ?>
-										<?php echo esc_html( siteorigin_setting( 'responsive_menu_text' ) ); ?>
-										<span class="screen-reader-text"><?php esc_html_e( 'Menu', 'siteorigin-north' ); ?></span>
-									<?php endif; ?>
-								</a>
+								<?php
+								$menu_id = ( wp_get_nav_menu_name( 'primary' ) ? wp_get_nav_menu_name( 'primary' ) :  false );
```
Comment (Misplon) [inline header.php]: There is an extra space before the `false`.

File: header.php [hunk 2]
```diff
@@ -85,15 +85,21 @@
...
+								if ( wp_get_nav_menu_object( $menu_id )->count || siteorigin_setting( 'navigation_search' ) ) : ?>
```
Comment (Misplon) [inline header.php]: If no menu is assigned to the primary menu location a notice is returned: `Notice: Trying to get property 'count' of non-object in /Users/andrew/Dropbox/Sites/siteorigin/wp-content/themes/siteorigin-north/header.php on line 90`.

**Comment by AlexGStapleton** [comment]
How does that look?

**Comment by Misplon** [comment]
Thanks

* Should we consider renaming `$show_min_cart` to `$show_mini_cart`, `min` makes me think minification or minimized. 

* The mini-cart is now only displaying in the mobile menu if the primary menu is empty.

* When the mini-cart does display in the mobile menu, it's displaying above the search, it should be below the search. Ref: https://demo.siteorigin.com/north/.

**Comment by AlexGStapleton** [comment]
> Should we consider renaming $show_min_cart to $show_mini_cart, min makes me think minification or minimized.

Oh sorry, that was a typo - that was always my intention.

How does that look?

**Comment by Misplon** [comment]
Super, thanks. The functionality looks good. Will take another quick look over it all in the morning.

File: header.php [hunk 3]
```diff
@@ -85,40 +85,51 @@
...
-									<?php siteorigin_north_display_icon( 'menu' ) ?>
-									<?php if ( siteorigin_setting( 'responsive_menu_text' ) ) : ?>
-										<?php echo esc_html( siteorigin_setting( 'responsive_menu_text' ) ); ?>
-										<span class="screen-reader-text"><?php esc_html_e( 'Menu', 'siteorigin-north' ); ?></span>
-									<?php endif; ?>
-								</a>
+								<?php
+								$menu_id = ( wp_get_nav_menu_name( 'primary' ) ? wp_get_nav_menu_name( 'primary' ) : false );
+								if ( class_exists( 'Woocommerce' ) ) {
+									// Work out if the WooComemrce cart is enabled and showing
```
Comment (Misplon) [inline header.php]: @AlexGStapleton Typo here. Thanks.

**Comment by Misplon** [comment]
If no menu is assigned to the primary menu position, the menu automatically populates. However, in this use case, the mobile menu is blank.

**Comment by AlexGStapleton** [comment]
Strangely, when no primary menu is assigned the `menu` class is applied to a different element. The last PR will account for both instances.

**Pull request merged**